File: Emisor.java
public class Emisor {
    public static void main(String[] args) {
        Aplicacion app = new Aplicacion();
        String mensaje = app.solicitarMensaje();

        Presentacion presentacion = new Presentacion();
        String binario = presentacion.codificarMensaje(mensaje);

        Enlace enlace = new Enlace();
        String mensajeConIntegridad = enlace.calcularIntegridad(binario);

        Ruido ruido = new Ruido();
        String mensajeConRuido = ruido.aplicarRuido(mensajeConIntegridad);

        System.out.println("Mensaje final con ruido: " + mensajeConRuido);
    }
}

File: Emisor.py
from Aplicacion import Aplicacion
from Presentacion import Presentacion
from Enlace import Enlace
from Ruido import Ruido

def main():
    aplicacion = Aplicacion()
    presentacion = Presentacion()
    enlace = Enlace()
    ruido = Ruido()

    mensaje = aplicacion.solicitar_mensaje()
    tasa_ruido = aplicacion.solicitar_tasa_de_ruido()

    binario = presentacion.codificar_mensaje(mensaje)
    print(f"Mensaje codificado en binario: {binario}")

    binario_con_paridad = enlace.calcular_integridad(binario)
    print(f"Mensaje con bit de paridad: {binario_con_paridad}")

    mensaje_con_ruido = ruido.aplicar_ruido(binario_con_paridad, tasa_ruido)
    print(f"Mensaje final con ruido: {mensaje_con_ruido}")

if __name__ == "__main__":
    main()
